### PR TITLE
add new variant

### PR DESCRIPTION
**Pull request by ladyada**

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -378,6 +378,55 @@ adafruit_qtpy_m0.menu.debug.off=Off
...
+adafruit_neotrinkey_m0.upload.native_usb=true
+adafruit_neotrinkey_m0.build.mcu=cortex-m0plus
+adafruit_neotrinkey_m0.build.f_cpu=48000000L
+adafruit_neotrinkey_m0.build.usb_product="NeoPixel Trinkey M0"
+adafruit_neotrinkey_m0.build.usb_manufacturer="Adafruit"
+adafruit_neotrinkey_m0.build.board=NEOTRINKEY_M0
+adafruit_neotrinkey_m0.build.core=arduino
+adafruit_neotrinkey_m0.build.extra_flags=-DCRYSTALLESS -DADAFRUIT_NEOTRINKEY_M0 -D__SAMD21E18A__ -DARM_MATH_CM0PLUS {build.usb_flags}
+adafruit_neotrinkey_m0.build.ldscript=linker_scripts/gcc/flash_with_bootloader.ld
+adafruit_neotrinkey_m0.build.openocdscript=openocd_scripts/trinket_m0.cfg
```
Comment (hathach) [inline boards.txt:405]: should be `openocd_scripts/neotrinkey_m0.cfg`

File: variants/neotrinkey_m0/variant.h [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+ * architecture. If you need to check if a pin supports PWM you must
+ * use digitalPinHasPWM(..).
+ *
+ * https://github.com/arduino/Arduino/issues/1833
+ */
+// #define digitalPinToTimer(P)
+
+// LEDs
+#define PIN_NEOPIXEL         (0u)
+
```
Comment (hathach) [inline variants/neotrinkey_m0/variant.h:81]: some variant did define `NEOPIXEL_NUM` (BLM badge) maybe we should define it as well.

**Review by hathach — COMMENTED** [review]
except for the minor typo for the openocd_scripts, everything else look great.

File: variants/neotrinkey_m0/variant.h (hunk 2 above)
Comment (ladyada) [reply variants/neotrinkey_m0/variant.h:81]: ok!

**Review by hathach — APPROVED** [review]
look perfect now :+1:

**Pull request merged**